src/evaluator.rs

```diff
@@ -18,10 +18,10 @@ impl Evaluator {
         &self.errors
     }
 
-    pub fn evaluate(&mut self, ast: &AbstractSyntaxTree) -> i32 {
+    pub fn evaluate(&mut self, ast: &'a AbstractSyntaxTree) -> i32 {
```

Comment:
Are we sure that one Evaluator will be used for more than one AST? Because if we are not sure, I think we could lose these lifetime annotations maybe?

Reply:
The analyzer made me add them. Feel free to remove if you can get it to build.